pypit/arms.py

```diff
@@ -287,80 +342,101 @@ def ARMS(spectrograph, fitstbl, setup_dict):
             except KeyError:  # No object extracted
                 continue
 
+        if len(all_specobjs) == 0:
+            msgs.warn('No objects!')
+            continue
+
         # Write 1D spectra
         save_format = 'fits'
         if save_format == 'fits':
-            outfile = settings.argflag['run']['directory']['science']+'/spec1d_{:s}.fits'.format(basename)
-            helio_dict = dict(refframe=settings.argflag['reduce']['calibrate']['refframe'],
+            outfile = os.path.join(_par['rdx']['scidir'], 'spec1d_{:s}.fits'.format(basename))
+            helio_dict = dict(refframe='pixel'
+                              if caliBrate.par['wavelengths']['reference'] == 'pixel' 
+                              else caliBrate.par['wavelengths']['frame'],
                               vel_correction=sci_dict['meta']['vel_corr'])
             arsave.save_1d_spectra_fits(all_specobjs, fitstbl[scidx], outfile,
-                                            helio_dict=helio_dict, obs_dict=settings.spect['mosaic'])
+                                        helio_dict=helio_dict, telescope=_spectrograph.telescope)
         elif save_format == 'hdf5':
-            debugger.set_trace()  # NEEDS REFACTORINGj
+            debugger.set_trace()  # NEEDS REFACTORING
```

Comment:
Maybe raise a warning here just before the debugger?

Reply:
I've just commented this out for the time being.  That part of the code is never touched because a few lines above the code says:
```
save_format = 'fits'
if save_format == 'fits':
```